modules/drivers/mongo/src/metabase/driver/mongo/query_processor.clj

```diff
@@ -363,8 +368,15 @@
 (defmethod ->rvalue :concat    [[_ & args]] {"$concat" (mapv ->rvalue args)})
 (defmethod ->rvalue :substring [[_ & args]] {"$substrCP" (mapv ->rvalue args)})
 
-(defmethod ->rvalue :+ [[_ & args]] {"$add" (mapv ->rvalue args)})
-(defmethod ->rvalue :- [[_ & args]] {"$subtract" (mapv ->rvalue args)})
+(defmethod ->rvalue :+ [[_ & [lhs rhs :as args]]]
+  (cond
+    (= (first lhs) :interval) (date-op "$dateAdd" rhs lhs)
```

Comment:
Is this order valid in MBQL?

Reply:
Conceptually it seems like it should be valid, but that schema is probably overly strict. I would still try to support it here so when we fix the schema we won't have to go back and fix this too.

Dumb question: wouldn't it make more sense to just implement `->rvalue` for `:interval`? Because otherwise this wouldn't fix `[:- <whatever> [:interval ...]]` forms. Only `:+`

Reply:
Oh nevermind I guess that wouldn't work because we have to use `$dateAdd` instead of `$add`. Well either way it would still be good to try to support this for `:-` forms as well. Better to fix it while we're poking around this part of the codebase instead of it getting opened as an issue in the future